File: itcloud/src/test/java/com/itinfo/itcloud/service/ItHostServiceTest.java
package com.itinfo.itcloud.service;

import com.itinfo.itcloud.model.computing.ClusterVo;
import com.itinfo.itcloud.model.computing.HostVo;
import com.itinfo.itcloud.model.computing.NicVo;
import com.itinfo.itcloud.model.computing.VmVo;
import com.itinfo.itcloud.model.create.HostCreateVo;
import com.itinfo.itcloud.model.error.CommonVo;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
class ItHostServiceTest {
    @Autowired ItHostService hostService;

    String defaultId = "f08baae8-2137-490c-bec2-fd00f67a37b9";

    @Test
    @DisplayName("호스트 목록")
    void getList() {
        List<HostVo> hostList = hostService.getList();

        assertThat(hostList.size()).isEqualTo(2);
        hostList.forEach(System.out::println);
    }

    @Test
    @DisplayName("호스트 생성 클러스터 리스트 출력")
    void setHostDefaultInfo() {
        List<ClusterVo> result = hostService.setHostDefaultInfo();

        result.forEach(System.out::println);
    }

    @Test
    @DisplayName("호스트 생성")
    void addHost() {
        HostCreateVo create =
                HostCreateVo.builder()
                        .name("host01.ititinfo.com")
                        .comment("192.168.0.81")
                        .hostIp("host01.ititinfo.com")
                        .sshPw("adminRoot!@#")
                        .spm(1)
                        .sshPort(22)
//                        .hostEngine(false)
                        .clusterId("9c7452ea-a5f3-11ee-93d2-00163e39cb43")
                        .build();

        CommonVo<Boolean> result = hostService.addHost(create);

        assertThat(result.getHead().getCode()).isEqualTo(201);
    }

    @Test
    @DisplayName("호스트 수정창")
    void setHost() {
        HostCreateVo result = hostService.setHost(defaultId);
        
        System.out.println(result);
    }

    @Test
    void editHost() {
    }

    @Test
    @DisplayName("호스트 삭제")
    void deleteHost() {
        String id = "1c8ed321-28e5-4f83-9e34-e13f9125f253";
        CommonVo<Boolean> result = hostService.deleteHost(id);

        assertThat(result.getHead().getCode()).isEqualTo(200);
    }

    @Test
    @DisplayName("호스트 유지보수")
    void deActive() {
        String id = "1c8ed321-28e5-4f83-9e34-e13f9125f253";
        hostService.deActive(id);

    }

    @Test
    void active() {
    }

    @Test
    void refresh() {
    }

    @Test
    void reStart() {
    }

    @Test
    void stop() {
        String id = "507b4118-346a-47be-bd3a-a15abc36d26a";

        CommonVo<Boolean> result = hostService.stop(id);
        assertThat(result.getHead().getCode()).isEqualTo(200);
    }

    @Test
    void getInfo() {
    }

    @Test
    @DisplayName("호스트 가상머신 목록")
    void getVm() {
        String id = "507b4118-346a-47be-bd3a-a15abc36d26a";
        List<VmVo> vmVoList = hostService.getVm(id);

        vmVoList.forEach(System.out::println);
    }

    @Test
    @DisplayName("호스트 가상머신 실행")
    void startVm() {
        String vmId = "eec63849-5026-482c-8f05-1d8e419ef548";
        CommonVo<Boolean> result = hostService.startVm(vmId);

        assertThat(result.getHead().getCode()).isEqualTo(200);
    }

    @Test
    @DisplayName("호스트 가상머신 일시정지")
    void pauseVm() {
        String vmId = "eec63849-5026-482c-8f05-1d8e419ef548";
        CommonVo<Boolean> result = hostService.pauseVm(vmId);

        assertThat(result.getHead().getCode()).isEqualTo(200);
    }

    @Test
    @DisplayName("호스트 가상머신 종료")
    void shutdownVm() {
        String vmId = "eec63849-5026-482c-8f05-1d8e419ef548";
        CommonVo<Boolean> result = hostService.stopVm(vmId);

        assertThat(result.getHead().getCode()).isEqualTo(200);
    }

    @Test
    @DisplayName("호스트 가상머신 전원끔")
    void stopVm() {
        String vmId = "eec63849-5026-482c-8f05-1d8e419ef548";
        CommonVo<Boolean> result = hostService.shutdownVm(vmId);

        assertThat(result.getHead().getCode()).isEqualTo(200);
    }

    @Test
    @DisplayName("호스트 가상머신 마이그레이션")
    void migrationVm() {

    }

    @Test
    @DisplayName("호스트 가상머신 마이그레이션 취소")
    void migrationCancelVm() {

    }




    @Test
    void getNic() {
        String hostId = "1c8ed321-28e5-4f83-9e34-e13f9125f253";
        List<NicVo> nicList = hostService.getNic(hostId);

        nicList.forEach(System.out::println);
        assertThat(nicList.size()).isEqualTo(2);
    }

    @Test
    void getHostDevice() {
    }

    @Test
    void getPermission() {
    }

    @Test
    void setAffinityDefaultInfo() {
    }

    @Test
    void getAffinitylabels() {
    }

    @Test
    void addAffinitylabel() {
    }

    @Test
    void getAffinityLabel() {
    }

    @Test
    void editAffinitylabel() {
    }

    @Test
    void deleteAffinitylabel() {
    }

    @Test
    void getEvent() {
    }
}

File: itcloud/src/test/java/com/itinfo/itcloud/none/ItHostServiceTest.kt
package com.itinfo.itcloud.none

import com.itinfo.common.LoggerDelegate
import com.itinfo.itcloud.service.computing.ItHostService
import org.junit.jupiter.api.Test
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.not
import org.hamcrest.Matchers.nullValue
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

/**
 * [ItHostServiceTest]
 * [ItHostService]에 대한 단위테스트
 *
 * @author chanhi2000
 * @since 2024.03.05
 */
@SpringBootTest
class ItHostServiceTest {
	@Autowired
	private lateinit var service: ItHostService

	/**
	 * [should_getName]
	 * [ItHostService.getName]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getName]
	 */
	@Test
	fun should_getName() {
		log.debug("should_getName ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getList]
	 * [ItHostService.getList]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getList]
	 */
	@Test
	fun should_getList() {
		log.debug("should_getList ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getInfo]
	 * [ItHostService.getInfo]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getInfo]
	 */
	@Test
	fun should_getInfo() {
		log.debug("should_getInfo ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getVm]
	 * [ItHostService.getVm]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getVm]
	 */
	@Test
	fun should_getVm() {
		log.debug("should_getVm ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getNic]
	 * [ItHostService.getNic]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getNic]
	 */
	@Test
	fun should_getNic() {
		log.debug("should_getNic ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getHostDevice]
	 * [ItHostService.getHostDevice]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getHostDevice]
	 */
	@Test
	fun should_getHostDevice() {
		log.debug("should_getHostDevice ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getPermission]
	 * [ItHostService.getPermission]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getPermission]
	 */
	@Test
	fun should_getPermission() {
		log.debug("should_getPermission ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getAffinitylabels]
	 * [ItHostService.getAffinitylabels]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getAffinitylabels]
	 */
	@Test
	fun should_getAffinitylabels() {
		log.debug("should_getAffinitylabels ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getEvent]
	 * [ItHostService.getEvent]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getEvent]
	 */
	@Test
	fun should_getEvent() {
		log.debug("should_getEvent ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getClusterList]
	 * [ItHostService.getClusterList]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getClusterList]
	 */
	@Test
	fun should_getClusterList() {
		log.debug("should_getClusterList ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getHostCreate]
	 * [ItHostService.getHostCreate]에 대한 단위테스트
	 * 
	 * @see [ItHostService.getHostCreate]
	 */
	@Test
	fun should_getHostCreate() {
		log.debug("should_getHostCreate ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_addHost]
	 * [ItHostService.add]에 대한 단위테스트
	 * 
	 * @see [ItHostService.add]
	 */
	@Test
	fun should_addHost() {
		log.debug("should_addHost ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_editHost]
	 * [ItHostService.update]에 대한 단위테스트
	 * 
	 * @see [ItHostService.update]
	 */
	@Test
	fun should_editHost() {
		log.debug("should_editHost ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_rebootHost]
	 * [ItHostService.rebootHost]에 대한 단위테스트
	 * 
	 * @see [ItHostService.rebootHost]
	 */
	@Test
	fun should_rebootHost() {
		log.debug("should_rebootHost ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_deleteHost]
	 * [ItHostService.remove]에 대한 단위테스트
	 * 
	 * @see [ItHostService.remove]
	 */
	@Test
	fun should_deleteHost() {
		log.debug("should_deleteHost ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_deActive]
	 * [ItHostService.deActive]에 대한 단위테스트
	 * 
	 * @see [ItHostService.deActive]
	 */
	@Test
	fun should_deActive() {
		log.debug("should_deActive ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_active]
	 * [ItHostService.active]에 대한 단위테스트
	 * 
	 * @see [ItHostService.active]
	 */
	@Test
	fun should_active() {
		log.debug("should_active ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_refresh]
	 * [ItHostService.refresh]에 대한 단위테스트
	 * 
	 * @see [ItHostService.refresh]
	 */
	@Test
	fun should_refresh() {
		log.debug("should_refresh ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_reStart]
	 * [ItHostService.reStart]에 대한 단위테스트
	 * 
	 * @see [ItHostService.reStart]
	 */
	@Test
	fun should_reStart() {
		log.debug("should_reStart ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_start]
	 * [ItHostService.start]에 대한 단위테스트
	 * 
	 * @see [ItHostService.start]
	 */
	@Test
	fun should_start() {
		log.debug("should_start ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_stop]
	 * [ItHostService.stop]에 대한 단위테스트
	 * 
	 * @see [ItHostService.stop]
	 */
	@Test
	fun should_stop	() {
		log.debug("should_stop ...")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}


	companion object {
		private val log by LoggerDelegate()
	}
}